### PR TITLE
Enhancement: Initial Node Refactor

### PR DESCRIPTION
**Pull request by Tigh-Gherr**
Even though this is WIP, given the early stages of development for this lib, I think it should be merged in anyway given that repo is currently in an unusable state.

This is the initial refactor for the bitcoin node functionality, providing RPC command functionality. Example usage:

```go
c := bn.NewNodeClient(
    bn.WithHost("http://bitcoin-host:18332"),
    bn.WithCreds("username", "password"),
)

txID, err := c.SendToAddress(ctx, "address", 5.0, nil)
if err != nil {
    return err
}

blocks, err := c.Generate(ctx, 1, nil)
if err != nil {
    return err
}

tx, err := c.RawTransaction(ctx, txID)
if err != nil {
    return err
}
```

Where possible, I have opted to use our `go-bt` and `go-bc` structs, over the modelling out the node response, or returning raw hex. So `RawTransaction`, for example, returns a `*bt.Tx`, instead of a hex string, or node formatted json.

An example of this can be seen in the example of funding a tx as the node user, to send to a random address. Here, instead of returning the raw tx hex at any point in the structs, we swap it out with `*bt.Tx`:
```go
tx := bt.NewTx()
if err := tx.AddP2PKHOutputFromAddress("mpvmxsAgDBGfdfM7WCJXq3UrJxBpfoH2RC", 20000000000); err != nil {
	panic(err)
}
                                                                                                        
fundedTx, err := c.FundRawTransaction(ctx, tx, nil)
if err != nil {
	panic(err)
}
                                                                                                        
signedTx, err := c.SignRawTransaction(ctx, fundedTx.Tx, nil)
if err != nil {
	panic(err)
}
                                                                                                        
txID, err := c.SendRawTransaction(ctx, signedTx.Tx, nil)
if err != nil {
	panic(err)
}
```

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+## How to contribute to go-bt
+
+#### **Did you find a bug?**
+
+* **Do not open up a GitHub issue if the bug is a security vulnerability
+  in go-bt**, and instead to refer to our [security policy](https://github.com/libsv/libsv/blob/master/SECURITY.md).
```
Comment (jadwahab) [inline .github/CONTRIBUTING.md:6]: rename to go-bn

File: models/block.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		return err
+	}
+
+	b.Txs = txs
+	b.BlockHeader = bj.BlockHeader
+	return nil
+}
+
+// BlockHeader model.
+type BlockHeader struct {
```
Comment (jadwahab) [inline models/block.go:42]: do you wanna use the go-bc block header here for some of the fields like you did with the tx output?

File: control.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import (
+	"context"
+	"time"
+
+	"github.com/libsv/go-bn/models"
+)
+
+// ControlClient interfaces interaction with the control sub commands on a bitcoin node.
+type ControlClient interface {
+	ActiveZMQNotifications(ctx context.Context) ([]*models.ZMQNotification, error)
```
Comment (jadwahab) [inline control.go:12]: do you think we should separate the zmq stuff into their own thing? @Tigh-Gherr

**Review by jadwahab — COMMENTED** [review]
really good stuff mate!

File: control.go (hunk 3 above)
Comment (Tigh-Gherr) [reply control.go:12]: Yeah that's the plan, there will be a `zmq` sub directory or something.

Any reference to zmq at the comment are actually bitcoin node rpc functions, so no actual zmq work has been done yet. That's this weeks agenda.

File: models/block.go (hunk 2 above)
Comment (Tigh-Gherr) [reply models/block.go:42]: This struct here has extra fields, if you're happy to lose them then yeah I can move this over

Comment (jadwahab) [reply models/block.go:42]: i meant have the bc header plus these extra fields (like you did here: https://github.com/libsv/go-bn/pull/1/files#diff-6d5301c97541b896843d5363919d90686ed3489ab0ced2d2977848ff6daafb74R17)

Comment (Tigh-Gherr) [reply models/block.go:42]: Lol I didn't even remember doing that. I'll move that there, good find

**Pull request merged**